presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/UnaliasSymbolReferences.java

```diff
@@ -592,19 +594,19 @@ public PlanNode visitIndexJoin(IndexJoinNode node, RewriteContext<Void> context)
         @Override
         public PlanNode visitUnion(UnionNode node, RewriteContext<Void> context)
         {
-            return new UnionNode(node.getId(), rewriteSources(node, context).build(), canonicalizeSetOperationVariableMap(node.getVariableMapping()));
+            return new UnionNode(node.getId(), rewriteSources(node, context).build(), fromListMultimap(canonicalizeSetOperationVariableMap(toListMultimap(node.getVariableMapping()))));
```

Comment:
Instead of convert it back/forth, probably just update `canonicalizeSetOperationVariableMap` to return `Map<VariableReferenceExpression, List<VariableReferenceExpression>>`?